docs/firebase_in_libraries.md

```diff
@@ -0,0 +1,53 @@
+# Using Firebase from a Framework or a library
+
+Currently all source code and binary Firebase SDKs are compiled as **static** frameworks.
+It may be not too important detail if Firebase SDKs are used from an app target directly,
+but must be taken into account if Firebase is used from another library or a framework.
+The key difference is in the way static and dynamic linking works. Your framework itself may
+be either static or dynamic. Let's consider these two options in more details.
+
+## Using Firebase SDKs from dynamic framework
+
+When a dynamic framework/library is compiled, each static dependency symbols
+are added to the dynamic framework binary. It is not really important unless you need to
+use both the dynamic framework and one of its static dependencies directly in your app or another framework. In this case if you attempt to link
+the same static framework/library to both the app and the dynamic framework you will end up with the static framework symbols added to both the app and the dynamic framework binaries. It means that when you app launches
+it will have two copy of the static framework symbols. This leads to undefined behavior (especially when
+different versions of the static framework are linking to the app and the dynamic framework). For example, a `dispatch_once` may or may not do the right initialization since there are now two entities to initialize. Here are couple more examples of issues related to this undefined behavior: #4315, #5152, etc.
```

Comment:
Looks like GitHub links don't work from markdown. Please make into markdown links.

Also keep line lengths < 100 so the md file is readable in code editors